### PR TITLE
Move RAF check to first render()

### LINKED ISSUE #11114: Check for rAF polyfill lazily
Cases like https://github.com/facebookincubator/create-react-app/issues/3199#issuecomment-334475000 can be pretty confusing. We should probably move the warning to happen during first `ReactDOM.render` rather than immediately.

### PR DESCRIPTION
**Pull request by jquense**
fixes #11114

### CODE REVIEW

File: src/renderers/dom/fiber/ReactDOMFiberEntry.js [hunk 1]
```diff
@@ -731,6 +749,7 @@ var ReactDOMFiber = {
     container: DOMContainer,
     callback: ?Function,
   ) {
+    checkForRAF();
     return renderSubtreeIntoContainer(
```
Comment (gaearon) [inline src/renderers/dom/fiber/ReactDOMFiberEntry.js]: Can we move into `renderSubtreeIntoContainer` since it covers more code paths?

**Comment by gaearon** [comment]
Can you verify if it fixes the setup in https://github.com/facebookincubator/create-react-app/issues/3199#issuecomment-334475000 in practice?

**Comment by sebmarkbage** [comment]
Has there been any discussion/arguments for why this is a good idea? Didn't see any in either issue or thread.

It's nice to cache these locally so that the linking is strong instead of having to do a global map look up on every invocation. In the Prepacked bundles we might want to cache this eagerly as part of the wiring up (and certainly in a wasm/asm bundle).

**Comment by gaearon** [comment]
The main issue is that the polyfill we recommended (patching global) doesn't work for people setting up Enzyme adapter because they do it earlier.

I guess we should just change our recommendation to some small package people can import.

**Comment by jquense** [comment]
I have to say I jumped on this because I spent an annoying amount of time the other day trying to get the warning to go away in jest only to realize it was enzyme, and had to awkwardly change imports to require to avoid hoisting issues or Yet Another Jest Setup File

**Comment by gaearon** [comment]
I figure we decided not to do it.

**Pull request closed without merging**